app/services/data/mock/generate-bill-run.service.js

```diff
@@ -0,0 +1,190 @@
+      ])
+    })
+    .withGraphFetched('billingInvoices.billingInvoiceLicences.billingTransactions.chargeElement.chargePurposes')
+    .modifyGraph('billingInvoices.billingInvoiceLicences.billingTransactions.chargeElement.chargePurposes', (builder) => {
+      builder.select([
+        'chargePurposeId',
+        'abstractionPeriodStartDay',
+        'abstractionPeriodStartMonth',
+        'abstractionPeriodEndDay',
+        'abstractionPeriodEndMonth',
+        'authorisedAnnualQuantity'
+      ])
+    })
+    .withGraphFetched('billingInvoices.billingInvoiceLicences.billingTransactions.chargeElement.chargePurposes.purposesUse')
+    .modifyGraph('billingInvoices.billingInvoiceLicences.billingTransactions.chargeElement.chargePurposes.purposesUse', (builder) => {
+      builder.select([
+        'description'
+      ])
+    })
+}
+
+/**
+ * Masks an invoice account number by replacing the first 3 digits, for example, T88898349A becomes Z11898349A
+ */
+function _maskInvoiceAccountNumber (invoiceAccountNumber) {
+  return `Z11${invoiceAccountNumber.substring(3)}`
+}
+
+/**
+ * Masks an invoice number by replacing the first 2 digits, for example, TAI0000011T becomes ZZI0000011T
+ */
+function _maskInvoiceNumber (invoiceNumber) {
+  return `ZZ${invoiceNumber.substring(2)}`
+}
+
+function _mockBillingInvoices (billingInvoices) {
+  billingInvoices.forEach((billingInvoice) => {
+    const { address, name } = GenerateMockDataService.go()
+
+    billingInvoice.accountAddress = address
+    billingInvoice.contact = name
+
+    billingInvoice.invoiceAccountNumber = _maskInvoiceAccountNumber(billingInvoice.invoiceAccountNumber)
+    billingInvoice.invoiceNumber = _maskInvoiceNumber(billingInvoice.invoiceNumber)
+
+    _mockBillingInvoiceLicences(billingInvoice.billingInvoiceLicences)
+  })
+}
+
+function _mockBillingInvoiceLicences (billingInvoiceLicences) {
+  billingInvoiceLicences.forEach((billingInvoiceLicence) => {
+    const { name } = GenerateMockDataService.go()
+    const { credit, debit, netTotal } = _transactionTotals(billingInvoiceLicence.billingTransactions)
+
+    billingInvoiceLicence.licenceHolder = name
+    billingInvoiceLicence.credit = credit
+    billingInvoiceLicence.debit = debit
+    billingInvoiceLicence.netTotal = netTotal
+  })
+}
+
+function _response (mockedBillingBatch) {
+  return MockBillRunPresenter.go(mockedBillingBatch)
+}
+
+/**
+ * Calculate the totals for a licence based on the transaction values
+ *
+ * We don't hold totals in the `billing_invoice_licence` record. But the UI shows them. We found it is calculating
+ * these on the fly in the UI code so we need to replicate the same behaviour.
+ *
+ * Another thing to note is that if a transaction is flagged as a credit, then `netAmount` will be held as a signed
+ * value, for example -213.40. This is why it might look confusing we are always adding on each iteration but the
+ * calculation will be correct.
+ * @param {*} transactions
+ * @returns
```

Comment:
I think I'd want to see us either flesh out `@param` with the type and `@returns` with some details, or remove them completely.